spec/features/suggested_salary_searches_spec.rb

```diff
@@ -0,0 +1,287 @@
+      .to_return(
+        body: course_fixture_for(results: number_of_results),
+        headers: { "Content-Type": "application/vnd.api+json; charset=utf-8" },
+    )
+  end
+
+  def stub_suggested_across_england_without_salary_filter(number_of_results:)
+    stub_request(:get, courses_url)
+      .with(query: suggested_search_count_parameters)
+      .to_return(
+        body: course_fixture_for(results: number_of_results),
+        headers: { "Content-Type": "application/vnd.api+json; charset=utf-8" },
+    )
+  end
+
+  def stub_results_with_salary_filter(radius:, number_of_results:)
+    stub_request(:get, courses_url)
+      .with(query: results_page_parameters.merge("filter[funding]" => "salary", "filter[latitude]" => 51.4980188, "filter[longitude]" => -0.1300436, "filter[radius]" => radius))
+      .to_return(
+        body: course_fixture_for(results: number_of_results),
+        headers: { "Content-Type": "application/vnd.api+json; charset=utf-8" },
+    )
+  end
+
+  def stub_suggested_with_salary_filter(radius:, number_of_results:)
+    stub_request(:get, courses_url)
+      .with(query: suggested_search_count_parameters.merge("filter[funding]" => "salary", "filter[latitude]" => 51.4980188, "filter[longitude]" => -0.1300436, "filter[radius]" => radius))
+      .to_return(
+        body: course_fixture_for(results: number_of_results),
+        headers: { "Content-Type": "application/vnd.api+json; charset=utf-8" },
+    )
+  end
+
+  def stub_suggested_without_salary_filter(radius:, number_of_results:)
+    stub_request(:get, courses_url)
+      .with(query: suggested_search_count_parameters.merge("filter[latitude]" => 51.4980188, "filter[longitude]" => -0.1300436, "filter[radius]" => radius))
+      .to_return(
+        body: course_fixture_for(results: number_of_results),
+        headers: { "Content-Type": "application/vnd.api+json; charset=utf-8" },
+    )
+  end
+
+  def search_for_salaried_courses_with_query_params(query)
+    filter_page.load(query: query)
+    filter_page.salary_courses.click
+    filter_page.find_courses.click
+  end
+
+  before do
+    stub_subjects
+  end
+
+  context "with 0 results" do
+    # TODO
+    # Edge cases
+    # Across england
+    # Moar distances
+
+    context "And the initial search was filtered to a 5 mile radius" do
+      before do
+        stub_results_with_salary_filter(radius: 5, number_of_results: 0)
+      end
+
+      context "with courses available that are non-salaried with the same radius and salaried in a larger radius" do
+        before do
+          stub_suggested_without_salary_filter(radius: 5, number_of_results: 4)
+          stub_suggested_with_salary_filter(radius: 10, number_of_results: 10)
+        end
+
+        it "displays a suggested search for the non-salaried courses with the same radius first followed with salaried in a larger radius" do
+          search_for_salaried_courses_with_query_params(default_query_for_location_search(radius: 5))
```

Comment:
Where we going to start writing these in more of the style that Apply are using? With the expectations as method calls? This is really readable IMO, just asking.